android/src/main/java/com/visioncamerapluginbarcodescanner/VisionCameraBitmapUtils.java

```diff
@@ -0,0 +1,140 @@
+
+    ScriptIntrinsicYuvToRGB scriptYuvToRgb = ScriptIntrinsicYuvToRGB.create(rs, Element.U8_4(rs));
+    scriptYuvToRgb.setInput(allocationYuv);
+    scriptYuvToRgb.forEach(allocationRgb);
+
+    allocationRgb.copyTo(bitmap);
+    return bitmap;
+  }
+
+  // https://stackoverflow.com/questions/47498843/incorrect-image-converting-yuv-420-888-into-bitmaps-under-android-camera2/47601824#47601824
+  private ByteBuffer imageToByteBuffer(final Image image) {
+    final Rect crop = image.getCropRect();
+    final int width = crop.width();
+    final int height = crop.height();
+
+    final Image.Plane[] planes = image.getPlanes();
+    final byte[] rowData = new byte[planes[0].getRowStride()];
+    final int bufferSize = width * height * ImageFormat.getBitsPerPixel(ImageFormat.YUV_420_888) / 8;
+    final ByteBuffer output = ByteBuffer.allocateDirect(bufferSize);
+
+    int channelOffset = 0;
+    int outputStride = 0;
+
+    for (int planeIndex = 0; planeIndex < 3; planeIndex++) {
+      if (planeIndex == 0) {
+        channelOffset = 0;
+        outputStride = 1;
+      } else if (planeIndex == 1) {
+        channelOffset = width * height + 1;
+        outputStride = 2;
+      } else if (planeIndex == 2) {
+        channelOffset = width * height;
+        outputStride = 2;
+      }
+
+      final ByteBuffer buffer = planes[planeIndex].getBuffer();
+      final int rowStride = planes[planeIndex].getRowStride();
+      final int pixelStride = planes[planeIndex].getPixelStride();
+
+      final int shift = (planeIndex == 0) ? 0 : 1;
+      final int widthShifted = width >> shift;
+      final int heightShifted = height >> shift;
+
+      buffer.position(rowStride * (crop.top >> shift) + pixelStride * (crop.left >> shift));
+
+      for (int row = 0; row < heightShifted; row++) {
+        final int length;
+
+        if (pixelStride == 1 && outputStride == 1) {
+          length = widthShifted;
+          buffer.get(output.array(), channelOffset, length);
+          channelOffset += length;
+        } else {
+          length = (widthShifted - 1) * pixelStride + 1;
+          buffer.get(rowData, 0, length);
+
+          for (int col = 0; col < widthShifted; col++) {
+            output.array()[channelOffset] = rowData[col * pixelStride];
+            channelOffset += outputStride;
+          }
+        }
+
+        if (row < heightShifted - 1) {
+          buffer.position(buffer.position() + rowStride - length);
+        }
+      }
+    }
+
+    return output;
+  }
+
+  private Bitmap getResizedBitmap(Bitmap image, int maxSize) {
+    int width = image.getWidth();
+    int height = image.getHeight();
+
+    float bitmapRatio = (float) width / (float) height;
+    if (bitmapRatio > 1) {
+      width = maxSize;
+      height = (int) (width / bitmapRatio);
+    } else {
+      height = maxSize;
+      width = (int) (height * bitmapRatio);
+    }
+    return Bitmap.createScaledBitmap(image, height, width , true);
```

Comment:
I think this is wrong. The 2nd parameter is width. So we might be swapping height/width here.